lms/djangoapps/completion/models.py

```diff
@@ -106,6 +106,81 @@ def submit_completion(self, user, course_key, block_key, completion):
         return obj, isnew
 
 
+class AggregateCompletionManager(models.Manager):
+    """
+    Custom manager for AggregateCompletion model.
+
+    """
+
+    def submit_completion(self, user, course_key, block_key, earned, possible):
+        """
+        Inserts and Updates the completion AggregateCompletion for the specified record.
+
+        Parameters:
+            * user (django.contrib.auth.models.User): The user for whom the
+              completion is being submitted.
+            * course_key (opaque_keys.edx.keys.CourseKey): The course in
+              which the submitted block is found.
+            * block_key (opaque_keys.edx.keys.UsageKey): The block that has had
+              its completion changed.
+            * earned: The fractional completion
+              value of the block (0.0 = incomplete, 1.0 = complete).
+            * possible: The fractional completion
+              value of the block (0.0 = incomplete, 1.0 = complete).
```

Comment:
possible: the sum of the possible completion values of all descendant blocks that are visible to the user. A positive integer.